examples/benchmark.py

```diff
@@ -81,7 +82,13 @@ def source(t, f0):
 data.set_receiver_pos(receiver_coords)
 data.set_shape(nt, 30)
 # A Forward propagation example
-jit_obj = AcousticWave2D_cg(model1, data, create_dm, nbpml=nbpml, t_order=2, s_order=2)
+
+# path at which numpy memmap file are stored
+cache_disk_path = "/tmp/devito_disk"
+if not os.path.exists(cache_disk_path):
+    os.makedirs(cache_disk_path)
+
+jit_obj = AcousticWave2D_cg(model1, data, create_dm, nbpml=nbpml, t_order=2, s_order=2, disk_path=cache_disk_path)
```

Comment:
Surely, setting an output filepath is something the user should do by simply passing this in as a variable, ie. `u = TimeData(..., filepath='path/to/my/file')`, and not via configuration files or env vars. What would otherwise happen if we have multiple time variables that spill to multiple files?

That being said, I believe we are missing the bigger picture here with this whole config discussion, which (at least to my mind) is quite simply that we have two layers:
* 1. Devito itself acts as a library-like layer which provides the tools (eg, `TimeData`, `Operator`, etc.) for developers to build applications that are based on solving PDEs elegantly by defining them symbolically and associating them with input data. This core part of devito should follow general library design principles, which is it should be highly configurable at run-time and not enforce any particular workflow on it's users (ie. "Why do you want me to drop out of my IPython shell so badly?"). 
* 2. The FWI/TTI examples on the other hand are just that: examples of one particular user application that uses the abstractions provided by devito. They should really live in the outer `inversion` repo and hopefully will be migrated there soon. Here all the domain-specific knowledge is encoded, including readers for all types of funky file formats. And for that layer I quite frankly don't care how you configure them and where you get the filepaths from, since all the application has to do is simply propagate that filepath to the right devito object (in our case probably by setting it on `u`).
* The one exception to the above points, and I believe maybe this is where the confusion started, are, to my mind, architecture-specific configurations, like compiler flags and settings. These should be configured through env vars (unless explicitly set in the application code), in order to provide platform-portability. That means I should be able to move a fully configured FWI-Forward operator code that I wrote on a Mac to a different platform (ie. some HPC cluster), simply by setting one or maybe two env flags to tell the JIT engine to use a specific compiler setting, ideally without touching the Python code.

In short, I am not opposed to using configure files or any other means of setting any application-level parameters, as long as they are confined to the application layer (in this case the specific `AcousticWave2D` class in examples). What I do not want is to have to write 20 lines into an obscure devito configuration before doing a live demonstration to a student showing how I can solve a simple diffusion PDE in only two or three lines in an IPython shell.